lib/ApplicationFeatures/ApplicationServer.cpp

```diff
@@ -52,7 +52,8 @@ ApplicationServer::ApplicationServer(std::shared_ptr<ProgramOptions> options,
     : _state(ServerState::UNINITIALIZED),
       _options(options),
       _stopping(false),
-      _binaryPath(binaryPath) {
+      _binaryPath(binaryPath),
+      _unpreparing(false) {
```

Comment:
We shouldn't add a new boolean control variable, as it gets more and more confusing with it.
The ApplicationServer already has a `_state` variable for the current state it is in. It already has an additional `_stopping` variable, which we may or may not be redundant. But adding a third state variable does not make it any better from may point of view. 
Can't we just use the already existing `_state` instead of `_unpreparing`?